ledger_device_sdk/src/nbgl.rs

```diff
@@ -59,13 +81,32 @@ impl From<SyncNbgl> for u8 {
             SyncNbgl::UxSyncRetRejected => 0x01,
             SyncNbgl::UxSyncRetQuitted => 0x02,
             SyncNbgl::UxSyncRetApduReceived => 0x03,
+            SyncNbgl::UxSyncRetSkipped => 0x04,
+            SyncNbgl::UxSyncRetContinue => 0x05,
+            SyncNbgl::UxSyncRetPinValidated => 0x06,
+            SyncNbgl::UxSyncRetPinRejected => 0x07,
             SyncNbgl::UxSyncRetError => 0xFF,
         }
     }
 }
 
 static mut G_RET: u8 = 0;
 static mut G_ENDED: bool = false;
+static mut G_CONFIRM_ASK_WHEN_TRUE: bool = false;
+static mut G_CONFIRM_ASK_WHEN_FALSE: bool = false;
+static mut G_CONFIRM_MESSAGE_WHEN_TRUE: Option<CString> = None;
+static mut G_CONFIRM_SUBMESSAGE_WHEN_TRUE: Option<CString> = None;
+static mut G_CONFIRM_OK_TEXT_WHEN_TRUE: Option<CString> = None;
+static mut G_CONFIRM_KO_TEXT_WHEN_TRUE: Option<CString> = None;
+static mut G_CONFIRM_MESSAGE_WHEN_FALSE: Option<CString> = None;
+static mut G_CONFIRM_SUBMESSAGE_WHEN_FALSE: Option<CString> = None;
+static mut G_CONFIRM_OK_TEXT_WHEN_FALSE: Option<CString> = None;
+static mut G_CONFIRM_KO_TEXT_WHEN_FALSE: Option<CString> = None;
```

Comment:
Not a big fan of the statics here; each of them will occupy memory even when not used (the actual string content is in the heap, but the `Option<CString>` still takes at least 12 bytes for pointer, length and capacity, possibly more).

A solution might be a single enum with an entry for each of the possible screens, where each entry contains exactly the required fields for the currently shown screen. Putting the enum in a single `static mut G_SCREEN_STRINGS: Box<UiScreenStrings>` should address this, and also group together strings that are used together, which might be more maintainable.

(The current space occupation is maybe not yet problematic, so you might want to do that in a separate PR if you prefer.)

Reply:
Yes good proposal, do you mean like this ?
```
pub(crate) enum ConfirmChoiceStrings {
    Undefined,
    WhenTrue {
        message: Option<CString>,
        submessage: Option<CString>,
        ok_text: Option<CString>,
        ko_text: Option<CString>,
    },
    WhenFalse {
        message: Option<CString>,
        submessage: Option<CString>,
        ok_text: Option<CString>,
        ko_text: Option<CString>,
    },
}
static mut G_CONFIRM_STRINGS: Box<ConfirmChoiceStrings> = Box::new(ConfirmChoiceStrings::Undefined);
```